astronomy-pipeline/README.md

```diff
@@ -2,7 +2,97 @@
+    - DB_NAME - the name of your database
+    - DB_PORT - port to access RDS (typically 5432 for postgres)
+    - ASTRONOMY_ID - obtained when signing up for the astronomy API
+    - ASTRONOMY_SECRET - secret key used for accessing the astronomy API
+
+### To Run on AWS:
+
+3. Create an ECR repository through terraform or the AWS UI.
+In order for your provisioned architecture one must dockerise their scripts and dependencies and push to an ECR repository.  
+For the next steps you will require AWS credentials and the ECR URI.
+
+4. Download the aws-cli:
+```bash
+brew install awscli
+```
+5. Verify your credentials:
+```bash
+aws configure
+```
+This will require sensitive information to be entered through the command line
+
+6. Add to your `.env` file the following:
+    - AWS_REGION - region of your ECR repository
+    - ECR_URI - endpoint of the ECR repository
+    - IMAGE_NAME - a name given to your docker image
+
+7. Run the following command to automatically dockerise and push to ECR:
+```bash
+bash deploy.sh
+```
+8. Now follow the terraform setup steps in the `../terraform` folder to provision a lambda based off this docker image.
 
 ## Usage
+- To use locally one can use the following command:
+```bash
+python pipeline.py
+```
+It is a good idea to ensure that the pipeline works with your credentials prior to provisioning the AWS architecture
+
+- To run tests run the following command:
+```bash
+pytest
+```
+Alternatively to run specific tests or test files:
+```bash
+pytest path/to/test_file.py
+pytest path/to/test_file.py::test_function_name
+```
+To get a coverage report execute the following;
+```bash
+pytest --cov
+```
 
-## How it works
\ No newline at end of file
+## How it works
+#### `astronomy_extract.py`
+- Orchestrates the extract portion of the pipeline
+- Uses `json `and `datetime` to partially process the data before transform steps
+#### `astronomy_extract_functions.py`
+- Encodes the `auth_string` which is necessary for accessing the API
+- Uses `psycopg2` to obtain the UK region coordinates
+- Uses `requests` to send a get request for celestial body data from the Astronomy API, as well as, moon phase data.
+#### `astronomy_transform.py`
+- Orchestrates the transform portion of the pipeline
+#### `astronomy_transform_functions.py`
+- Utilises `pandas`, `datetime` and `json` to create a dataframe and process the obtained data
+#### `astronomy_load.py`
+- Uploads the list generated from the `pandas` dataframe to the project database using `psycopg2`
+#### `astronomy_pipeline.py`
+- Written to conform to AWS lambda conventions
+- Orchestrates entire ETL pipeline
+#### `test_astronomy_[filename].py`
+- Contains the tests for the files within this folder
+- Uses `pytest` and `unittest.mock` for each test
+#### `deploy.sh`
+- Runs a series of commands that dockerise the pipeline using the local `Dockerfile`.
+- Pushes the docker image to a given ECR repository
+#### `db_connect.sh`
```

Comment:
Good consistent formatting in having these file name headings all next to one another.